pulp_container/app/cache.py

```diff
@@ -72,3 +73,19 @@ def find_base_path_cached(request, cached):
         except ObjectDoesNotExist:
             raise RepositoryNotFound(name=path)
         return distro.base_path
+
+
+class FlatpakCache(SyncContentCache):
```

Comment:
The `base-key` for the cache should (normally) be based on the base-path of the distribution for the request being made. See https://github.com/pulp/pulpcore/blob/main/pulpcore/app/util.py#L424 for how we normally determine the `base-key`. And then the `key` should typically be based on the incoming request (path/headers/method/etc...). 

This case seems to be special since this new endpoint is not based on the `base_path` of a distribution and seems to be serving content from multiple distributions. I think having it set to '/static/index' is fine. You'll need to update the hook on the container distribution and repository models to delete this cache entry upon update of those models. [0] [1]

Also, since the response of this endpoint changes based on the permissions of the user I think you need to add `request.user` to the `key` in `make_key`.

[0] https://github.com/pulp/pulpcore/blob/3851472b3c9bb27578729943040fca57fc52182b/pulpcore/app/models/publication.py#L601
[1] https://github.com/pulp/pulpcore/blob/3851472b3c9bb27578729943040fca57fc52182b/pulpcore/app/models/repository.py#L278

Reply:
@gerrod3, re "You'll need to update the hook on the container distribution and repository models":  Just to make sure:  The hook conditions on the added `ContainerDistribution` and `ContainerRepository`  `invalidate_cache` methods would need to cover at least all of the existing hook conditions on the base `Distribution` and `Repository` `invalidate_cache` methods, right?  (Or else `invalidate_cache` would stop being called for derived instances on the missing conditions, I assume.  Which looks a bit brittle, that the hooks in `pulp_container` would need to be adapted whenever the underlying hooks in `pulpcore` change.)

Reply:
You should just be able to override the `invalidate_cache` method on those two models, call `super` first and then add your own cache invalidate call. The hooks should stay inplace (atleast that's how I think Python should work).

Reply:
Ach, ignore me.  I can just add some completely unrelated `invalidate_flatpak_cache` methods to the derived classes, with their own sets of hook conditions...